src/partition/src/multi_dim.rs

```diff
@@ -0,0 +1,201 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+use std::cmp::Ordering;
+use std::collections::HashMap;
+
+use datatypes::prelude::Value;
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+use store_api::storage::RegionNumber;
+
+use crate::error::{self, Result};
+use crate::expr::{Operand, PartitionExpr, RestrictedOp};
+use crate::PartitionRule;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct MultiDimPartitionRule {
+    partition_columns: Vec<String>,
+    // name to index of `partition_columns`
+    name_to_index: HashMap<String, usize>,
+    regions: Vec<RegionNumber>,
+    exprs: Vec<PartitionExpr>,
+}
+
+impl MultiDimPartitionRule {
+    pub fn new(
+        partition_columns: Vec<String>,
+        regions: Vec<RegionNumber>,
+        exprs: Vec<PartitionExpr>,
+    ) -> Self {
+        let name_to_index = partition_columns
+            .iter()
+            .enumerate()
+            .map(|(i, name)| (name.clone(), i))
+            .collect::<HashMap<_, _>>();
+
+        Self {
+            partition_columns,
+            name_to_index,
+            regions,
+            exprs,
+        }
+    }
+
+    fn find_region(&self, values: &[Value]) -> Result<RegionNumber> {
+        ensure!(
+            values.len() == self.partition_columns.len(),
+            error::RegionKeysSizeSnafu {
+                expect: self.partition_columns.len(),
+                actual: values.len(),
+            }
+        );
+
+        for (region_index, expr) in self.exprs.iter().enumerate() {
+            if self.evaluate_expr(expr, values)? {
+                return Ok(self.regions[region_index]);
+            }
+        }
+
+        // return the default region number
```

Comment:
So ,the default region number is always zero?

Reply:
Yep